### PR TITLE
Integration tests for F#

### PR DESCRIPTION
**Pull request by psfinaki**
The tests for now are in C#. That's because:
1) the integration testing framework API is subject to change 
2) the current API is very F#-unfriendly
3) we are still investigating this topic

Tests run in a separate CI leg:
image

image

### CODE REVIEW

**Comment by T-Gro** [comment]
This is from the CI run on this PR:

Passed!  - Failed:     0, Passed:     7, Skipped:     0, **Total**:     **7**, Duration: **7 m 17 s** - FSharp.Editor.**IntegrationTests**.dll (net472)


I don't think this is a good idea to have.
As of now, I am not convinced that the benefits for this repo outweigh the added time and added reason for failing CI.

**Review by T-Gro — CHANGES_REQUESTED** [review]
Passed!  - Failed:     0, Passed:     7, Skipped:     0, Total:     7, Duration: 7 m 17 s - FSharp.Editor.IntegrationTests.dll (net472)

**Comment by psfinaki** [comment]
@T-Gro everything is an "added reason for failing CI". If these proves to be consistently passing, this shouldn't be an obstacle in that sense.

Extra time is a different thing. A few minutes is probably something we can live with but yeah that's not scalable. I will look into what other repos do about integration tests in CI, we should basically follow. @shamwell any advice?

File: FSharp.Editor.sln [hunk 1]
```diff
@@ -1,71 +1,80 @@
...
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "FSharp.Compiler.Service", "src\Compiler\FSharp.Compiler.Service.fsproj", "{AD603EF2-FAC6-48D1-AAEB-A6CF898062A9}"
+Project("{6EC3EE1D-3C4E-46DD-8F32-0CC8E7565705}") = "FSharp.Compiler.Service", "src\Compiler\FSharp.Compiler.Service.fsproj", "{AD603EF2-FAC6-48D1-AAEB-A6CF898062A9}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "FSharp.Core", "src\FSharp.Core\FSharp.Core.fsproj", "{67DA0BF3-AAD3-47F4-9EC6-AD8EC532B5A9}"
+Project("{6EC3EE1D-3C4E-46DD-8F32-0CC8E7565705}") = "FSharp.Core", "src\FSharp.Core\FSharp.Core.fsproj", "{67DA0BF3-AAD3-47F4-9EC6-AD8EC532B5A9}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "FSharp.DependencyManager.Nuget", "src\FSharp.DependencyManager.Nuget\FSharp.DependencyManager.Nuget.fsproj", "{24399E68-9000-4556-BDDD-8D74A9660D28}"
+Project("{6EC3EE1D-3C4E-46DD-8F32-0CC8E7565705}") = "FSharp.DependencyManager.Nuget", "src\FSharp.DependencyManager.Nuget\FSharp.DependencyManager.Nuget.fsproj", "{24399E68-9000-4556-BDDD-8D74A9660D28}"
```
Comment (vzarytovskii) [inline FSharp.Editor.sln:12]: Do you know what's the reason guid changed for some projects?

File: eng/Build.ps1 [hunk 2]
```diff
@@ -613,6 +613,7 @@ try {
     if ($testVs -and -not $noVisualStudio) {
         TestUsingNUnit -testProject "$RepoRoot\vsintegration\tests\UnitTests\VisualFSharp.UnitTests.fsproj" -targetFramework $desktopTargetFramework -testadapterpath "$ArtifactsDir\bin\VisualFSharp.UnitTests\"
         TestUsingXUnit -testProject "$RepoRoot\vsintegration\tests\FSharp.Editor.Tests\FSharp.Editor.Tests.fsproj" -targetFramework $desktopTargetFramework -testadapterpath "$ArtifactsDir\bin\FSharp.Editor.Tests\FSharp.Editor.Tests.fsproj"
+        TestUsingXUnit -testProject "$RepoRoot\vsintegration\tests\FSharp.Editor.IntegrationTests\FSharp.Editor.IntegrationTests.csproj" -targetFramework $desktopTargetFramework -testadapterpath "$ArtifactsDir\bin\FSharp.Editor.IntegrationTests\"
```
Comment (vzarytovskii) [inline eng/Build.ps1:616]: If we want to run it in CI, it should be a separate leg and test switch, so it doesn't contribute more to existing runs.

**Review by vzarytovskii — CHANGES_REQUESTED** [review]

File: eng/Build.ps1 (hunk 2 above)
Comment (psfinaki) [reply eng/Build.ps1:616]: Yep yep, working on that

File: FSharp.Editor.sln (hunk 1 above)
Comment (psfinaki) [reply FSharp.Editor.sln:12]: Nope, but I think VS sometimes randomly changes project GUIDs, I think I should be able to revert it

File: eng/Build.ps1 (hunk 2 above)
Comment (psfinaki) [reply eng/Build.ps1:623]: Okay yep that's working.

File: FSharp.Editor.sln (hunk 1 above)
Comment (psfinaki) [reply FSharp.Editor.sln]: Reverted, doesn't seem to affect anything.

**Review by KevinRansom — APPROVED** [review]

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**